### PR TITLE
[WIP] [core] Adapted DefaultSystemChannelTypeProvider to changed ESH API

### PR DESCRIPTION
**Pull request by cweitkamp**
Based on PR #3423 
Solution for 1. question of this [discussion](https://github.com/eclipse/smarthome/pull/3754#issuecomment-311349822) 
Preperation for ferature #3756 

I repeat my 2. question of this [discussion](https://github.com/eclipse/smarthome/pull/3754#issuecomment-311349822) - This is more a general architecture design question: Why are the default system channels hard coded into their own class? Isn't it more useful to define a xml-file for them? Like in every binding?

IMHO the `DefaultSystemChannelTypeProvider.java` class is located wrong in package `org.eclipse.smarthome.core.thing`. Should be moved to package `org.eclipse.smarthome.core.thing.xml` now.

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by sjka** [comment]
> Isn't it more useful to define a xml-file for them?

The whole framework is intended to work without the XML stuff. So it should also work when not deploying the XML related bundles (and obviously also bindings with don't rely in XMLs...).

**Comment by cweitkamp** [comment]
Thanks. Now it is clear to me.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/DefaultSystemChannelTypeProvider.java [hunk 1]
```diff
@@ -100,110 +101,38 @@
...
     public DefaultSystemChannelTypeProvider() {
-        channelGroupTypes = Collections.emptyList();
-        channelTypes = Collections.unmodifiableCollection(
-                Arrays.asList(new ChannelType[] { SYSTEM_CHANNEL_SIGNAL_STRENGTH, SYSTEM_CHANNEL_LOW_BATTERY,
-                        SYSTEM_CHANNEL_BATTERY_LEVEL, SYSTEM_TRIGGER, SYSTEM_RAWBUTTON, SYSTEM_BUTTON }));
+        final Bundle bundle = FrameworkUtil.getBundle(DefaultSystemChannelTypeProvider.class);
 
+        add(bundle, SYSTEM_CHANNEL_SIGNAL_STRENGTH);
+        add(bundle, SYSTEM_CHANNEL_LOW_BATTERY);
+        add(bundle, SYSTEM_CHANNEL_LOW_BATTERY);
```
Comment (cweitkamp) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/DefaultSystemChannelTypeProvider.java]: double

File: bundles/core/org.eclipse.smarthome.core.thing/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -15,6 +15,7 @@ Import-Package: com.google.common.base,
  org.eclipse.smarthome.config.core.dto,
  org.eclipse.smarthome.config.core.status,
  org.eclipse.smarthome.config.core.validation,
+ org.eclipse.smarthome.config.xml,
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/META-INF/MANIFEST.MF]: Sorry, that won't work. As @sjka mentioned, the framework should also work without xml support and thus core.thing must not have a dependency on config.xml.

**Comment by cweitkamp** [comment]
Any suggestions how we can solve this in a better way? I don't like the idea to have duplicate lines of code in different places.

**Comment by kaikreuzer** [comment]
I didn't look into the details of what you exactly need there, but it might be possible to extract a common abstract base class that could be put in config.core, so that this would be the only dependency you need then.

**Comment by kaikreuzer** [comment]
@cweitkamp Are you waiting for some further and better suggestions or did you just put this aside for the moment?

**Comment by cweitkamp** [comment]
@kaikreuzer I put it on-hold. The current solution is working and this is a nice to have refactoring. If anyone else is interested to work on this he should feel free to do so.

**Pull request closed without merging**